### PR TITLE
feat: implement error notification as pr comment

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

fixes https://github.com/kiegroup/git-backporting/issues/123

## Description

Implement an optional notification mechanism on the original pull request for error tracking.

If enabled, the tool will create a new comment in the original PR (the one you tried to backport) with some infos of the error that occurred.

This is going to introduce other two inputs:
- [x] enable error notification, disabled by default
- [ ] (optional) comment body, with a custom `{{error}}` placeholder that can be replaced by the actual error.

### Checklist
- [x] Tests added if applicable.
- [x] Documentation updated if applicable.

### Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them _after approval_ or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

> **Note:** `dist/cli/index.js` and `dist/gha/index.js` are automatically generated by git hooks and gh workflows.

<details>
<summary>
First time here?
</summary>

This project follows [git conventional commits](https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13) pattern, therefore the commits should have the following format:

```
<type>(<optional scope>): <subject>
empty separator line
<optional body>
empty separator line
<optional footer>
```

Where the type must be one of `[build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test]`

> **NOTE**: if you are still in a `work in progress` branch and you want to push your changes remotely, consider adding `--no-verify` for both `commit` and `push`, e.g., `git push origin <feat-branch> --no-verify` - this could become useful to push changes where there are still tests failures. Once the pull request is ready, please `amend` the commit and force-push it to keep following the adopted git commit standard. 

</details>

<details>
<summary>
How to prepare for a new release?
</summary>

There is no need to manually update `package.json` version and `CHANGELOG.md` information. This process has been automated in [Prepare Release](./workflows/prepare-release.yml) *Github* workflow.

Therefore whenever enough changes are merged into the `main` branch, one of the maintainers will trigger this workflow that will automatically update `version` and `changelog` based on the commits on the git tree.

More details can be found in [package release](https://github.com/kiegroup/git-backporting/blob/main/README.md#package-release) section of the README.
</details>

### CODE REVIEW

**Comment by lampajr** [comment]
@earl-warren fyi

File: action.yml [hunk 1]
```diff
@@ -109,6 +109,11 @@ inputs:
     description: >
       Semicolon separated list of additional comments to be posted to the backported pull request
     required: false
+  no-squash:
```
Comment (earl-warren) [inline action.yml:112]:
```suggestion
  enable-err-notification:
```

**Review by earl-warren — CHANGES_REQUESTED** [review]
One tiny change otherwise looks great.

**Comment by lampajr** [comment]
> One tiny change otherwise looks great.

Thanks, copy&paste issue lol

I need to drop for today, but I plan to finalize this tomorrow :rocket:

**Comment by lampajr** [comment]
> Open points:
> 
>     * I don't really like the default comment message I set right now, any feedback is welcome

As first implementation let's go with `Backporting failed: ${{error}}`

>     * Make the comment message configurable, I think this is a good thing but it can be done in a followup PR

Will create separate issue to make the comment body configurable through templating
 
> TODOs:

* [ ]  Implement the Gitlab client
* [ ]  Update documentation
* [ ]  Try e2e / manual tests

Still need to try some real examples, did not have time to setup ad hoc test but will do asap.

**Comment by lampajr** [comment]
Ok I was able to test it with some real examples, e.g., https://github.com/lampajr/backporting-example/pull/66#issuecomment-2048379252

With latest two commits:
- Disable the comment if `dry-run` is enabled
- Updated the default comment to "The backport to <TARGET_BRANCJ> failed. Check the latest run for more details."

I decided to remove the error from the comment as it could become very huge if the conflicts are many, e.g., https://github.com/lampajr/backporting-example/pull/66#issuecomment-2048357951

**Comment by lampajr** [comment]
@earl-warren If you want to take a further look the PR is ready, otherwise I will go and merge!

PS: don't feel obligated, you already did a lot!!!

**Review by earl-warren — APPROVED** [review]
:tada:

**Pull request merged**